src/Provider/HttpKernelServiceProvider.php

```diff
@@ -0,0 +1,98 @@
+<?php
+
+namespace DI\Bridge\Silex\Provider;
+
+use DI\Bridge\Silex\CallbackInvoker;
+use DI\Bridge\Silex\CallbackResolver;
+use DI\Bridge\Silex\Controller\ControllerResolver;
+use DI\Bridge\Silex\ConverterListener;
+use DI\Bridge\Silex\MiddlewareListener;
+use Interop\Container\ContainerInterface;
+use Invoker\ParameterResolver\AssociativeArrayResolver;
+use Invoker\ParameterResolver\Container\TypeHintContainerResolver;
+use Invoker\ParameterResolver\ResolverChain;
+use Pimple\ServiceProviderInterface;
+use Silex\Api\EventListenerProviderInterface;
+use Symfony\Component\EventDispatcher\EventDispatcherInterface;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+
+/**
+ * @author Jacob Dreesen <jacob.dreesen@gmail.com>
+ */
+class HttpKernelServiceProvider implements ServiceProviderInterface, EventListenerProviderInterface
+{
+    /**
+     * @var ContainerInterface
+     */
+    private $container;
+
+    /**
+     * @var CallbackInvoker
+     */
+    private $callbackInvoker;
+
+    /**
+     * @param ContainerInterface $container
+     * @param CallbackInvoker    $callbackInvoker
+     */
+    public function __construct(ContainerInterface $container, CallbackInvoker $callbackInvoker)
+    {
+        $this->container = $container;
+        $this->callbackInvoker = $callbackInvoker;
+    }
+
+    public function register(\Pimple\Container $app)
+    {
+        // Override the controller resolver with ours.
+        $app['resolver'] = function ($app) {
+            return new ControllerResolver(
+                $app['phpdi.callable_resolver'],
+                new ResolverChain([
+                    new AssociativeArrayResolver,
+                    new TypeHintContainerResolver($this->container),
+                ])
+            );
+        };
+
+        // Override the callback resolver with ours.
+        $app['callback_resolver'] = function ($app) {
+            return new CallbackResolver(
+                $app,
+                $app['phpdi.callable_resolver']
+            );
+        };
+    }
+
+    public function subscribe(\Pimple\Container $app, EventDispatcherInterface $dispatcher)
+    {
+        // Remove the Silex listeners first
+        $this->removeSubscriber($dispatcher, \Silex\EventListener\MiddlewareListener::class);
+        $this->removeSubscriber($dispatcher, \Silex\EventListener\ConverterListener::class);
+
+        // And register ours instead
+        $dispatcher->addSubscriber(new MiddlewareListener($app, $this->callbackInvoker));
+        $dispatcher->addSubscriber(new ConverterListener($app['routes'], $app['callback_resolver'], $this->callbackInvoker));
+    }
+
+    /**
+     * Removes an event subscriber by its class name.
+     *
+     * @param EventDispatcherInterface $dispatcher
+     * @param string                   $subscriberClass
+     */
+    private function removeSubscriber(EventDispatcherInterface $dispatcher, $subscriberClass)
```

Comment:
Because [it calls `removeListener()`](https://github.com/symfony/event-dispatcher/blob/0343b2cedd0edb26cdc791212a8eb645c406018b/EventDispatcher.php#L153) with the given subscriber object internally, which only seems able to [remove the original subscribed object](https://github.com/symfony/event-dispatcher/blob/0343b2cedd0edb26cdc791212a8eb645c406018b/EventDispatcher.php#L111) that has been registered. And we don't have access to these objects to remove them this way.